Xamarin.Essentials/Calendars/Calendars.android.cs

```diff
@@ -405,6 +405,34 @@ static async Task<bool> PlatformUpdateCalendarEvent(CalendarEvent eventToUpdate)
             throw new ArgumentException("[Android]: Could not update appointment with supplied parameters");
         }
 
+        static async Task<bool> PlatformSetEventRecurrenceEndDate(string eventId, DateTimeOffset recurrenceEndDate)
+        {
+            await Permissions.RequestAsync<Permissions.CalendarWrite>();
+
+            var existingEvent = await GetEventByIdAsync(eventId);
+            CalendarEvent thisEvent;
+            if (existingEvent == null || string.IsNullOrEmpty(existingEvent.CalendarId))
+            {
+                return false;
+            }
+            else
+            {
+                thisEvent = await GetEventByIdAsync(eventId);
+            }
+
+            thisEvent.RecurrancePattern.EndDate = recurrenceEndDate;
+            thisEvent.RecurrancePattern.TotalOccurrences = null;
+
+            var eventUri = CalendarContract.Events.ContentUri;
+            var eventValues = SetupContentValues(thisEvent);
+
+            if (Platform.AppContext.ApplicationContext.ContentResolver.Update(eventUri, eventValues, $"{CalendarContract.Attendees.InterfaceConsts.Id}={eventId}", null) > 0)
+            {
+                return true;
```

Comment:
my preference would be to have the exception be in the if clause eg
if (! Platform.......)
{
  throw new exception
}
return true;

Reply:
Altered accordingly